aspnet/fundamentals/routing.rst

```diff
@@ -1,114 +1,298 @@
+
+Additional overloads of ``MapRoute`` accept values for ``constraints``, ``dataTokens``, and ``defaults``. These additional parameters of ``MapRoute`` are defined as type ``object``. The typical usage of these parameters is to pass an anonymously typed object, where the property names of the anonymous type match route parameter names.
+
+The following two examples create equivalent routes:
+
+.. code-block:: c#
 
-.. literalinclude:: routing/sample/RoutingSample/project.json
-  :dedent: 2
-  :language: javascript
-  :lines: 11-20
-  :emphasize-lines: 7
+    routes.MapRoute(
+        name: "default_route",
+        template: "{controller}/{action}/{id?}",
+        defaults: new { controller = "Home", action = "Index" });
+
+    routes.MapRoute(
+        name: "default_route",
+        template: "{controller=Home}/{action=Index}/{id?}");
+
+.. tip:: The inline syntax for defining constraints and defaults can be more convenient for simple routes. However, there are features such as data tokens which are not supported by inline syntax.
+
+.. review-required: changed template and add MVC controller sample
+
+This example demonstrates a few more features:
+
+.. code-block:: c#
+
+  routes.MapRoute(
+    name: "blog",
+    template: "Blog/{*article}",
+    defaults: new { controller = "Blog", action = "ReadArticle" });
+
+This template will match a URL path like ``/Blog/All-About-Routing/Introduction`` and will extract the values ``{ controller = Blog, action = ReadArticle, article = All-About-Routing/Introduction }``. The default route values for ``controller`` and ``action`` are produced by the route even though there are no corresponding route parameters in the template. Default values can be specified in the route template. The ``article`` route parameter is defined as a *catch-all* by the appearance of an asterix ``*`` before the route parameter name. Catch-all route parameters capture the remainder of the URL path, and can also match the empty string. Using the route above and a request to ``/Blog/All-About-Routing/Introduction``, the following code will display "All-About-Routing/Introduction":
+
+.. code-block:: c#
+
+  using Microsoft.AspNetCore.Mvc;
   
-Add routing to ``ConfigureServices`` in *Startup.cs*:
+  public class BlogController : Controller
+  {
+      public string ReadArticle(string article)
+      {
+          return article;
+      }
+  }
+
+.. review-required: removed leading / from template, changed to use IntRouteConstraint, MapRoute, not MapGet. If you want to use MapGet in lieu of MapRoute we need to say why. ``MapGet`` only matches routes for GET requests.
```

Comment:
I think the `Map[Verb]` extensions will need their own section. Probably a follow up to this PR since this is already so big.